### PR TITLE
Replace memset code

### PR DESCRIPTION
**Pull request by flaviojs**
In #937 I found that I couldn't use `ST::string` inside `MOUSE_REGION` because of `memset`.

This PR aims to replace all `memset` code with something else.

Not convinced memset is bad?
Here is an article with more problems: https://www.viva64.com/en/b/0360/

---

Fixes too much mix memory, possible uninitialized byte and warnings in `SoundMan.cc`.

Fixes initialization of `gusMapPathingData` in `FindStratPath`.

### CODE REVIEW

**Comment by flaviojs** [comment]
Reworked commits so that the C/C++ includes come last.

**Comment by flaviojs** [comment]
Hmm, this is gonna take a lot of commits grouping by function... maybe I should group them by filename instead.

**Comment by flaviojs** [comment]
Done, it's ready for review.

File: src/game/Laptop/BobbyR.cc [hunk 1]
```diff
@@ -444,7 +446,7 @@ static void InitBobbyRayNewInventory(void)
 	UINT16	usBobbyrIndex = 0;
 
 
-	memset( LaptopSaveInfo.BobbyRayInventory, 0, sizeof(STORE_INVENTORY) * MAXITEMS);
+	std::fill_n(LaptopSaveInfo.BobbyRayInventory, static_cast<size_t>(MAXITEMS), STORE_INVENTORY{});
```
Comment (selaux) [inline src/game/Laptop/BobbyR.cc]: Wont that result in the array filled with references to the same `STORE_INVENTORY{}` struct?

File: src/game/TacticalAI/FindLocations.cc [hunk 2]
```diff
@@ -552,7 +554,7 @@ INT16 FindBestNearbyCover(SOLDIERTYPE *pSoldier, INT32 morale, INT32 *piPercentB
 #if defined( _DEBUG ) && !defined( PATHAI_VISIBLE_DEBUG )
 	if (gfDisplayCoverValues)
 	{
-		memset( gsCoverValue, 0x7F, sizeof( INT16 ) * WORLD_MAX );
+		std::fill_n(gsCoverValue, WORLD_MAX, 0x7F7F);
```
Comment (selaux) [inline src/game/TacticalAI/FindLocations.cc]: What does `0x7F` actually stand for in this case?

File: src/game/TacticalAI/AIMain.cc [hunk 3]
```diff
@@ -69,7 +71,7 @@ void InitAI(void)
 #ifdef _DEBUG
 	if (gfDisplayCoverValues)
 	{
-		memset( gsCoverValue, 0x7F, sizeof( INT16 ) * WORLD_MAX );
+		std::fill_n(gsCoverValue, WORLD_MAX, 0x7F7F);
```
Comment (selaux) [inline src/game/TacticalAI/AIMain.cc]: What does `0x7F` actually stand for in this case?

File: src/game/Tactical/PathAI.cc [hunk 4]
```diff
@@ -719,13 +721,13 @@ INT32 FindBestPath(SOLDIERTYPE* s, INT16 sDestination, INT8 ubLevel, INT16 usMov
...
-	memset( pathQ, 0, iMaxPathQ * sizeof( path_t ) );
-	memset( trailTree, 0, iMaxTrailTree * sizeof( trail_t ) );
+	std::fill_n(pathQ, iMaxPathQ, path_t{});
+	std::fill_n(trailTree, iMaxTrailTree, trail_t{});
 
 #if defined( PATHAI_VISIBLE_DEBUG )
 	if (gfDisplayCoverValues && gfDrawPathPoints)
 	{
-		memset( gsCoverValue, 0x7F, sizeof( INT16 ) * WORLD_MAX );
+		std::fill_n(gsCoverValue, WORLD_MAX, 0x7F7F);
```
Comment (selaux) [inline src/game/Tactical/PathAI.cc]: What does `0x7F` actually stand for in this case?

File: src/game/Laptop/BobbyR.cc (hunk 1 above)
Comment (flaviojs) [reply src/game/Laptop/BobbyR.cc]: Nope, `LaptopSaveInfo.BobbyRayInventory` is an array of concrete structures, not references.
`STORE_INVENTORY{}` is a temporary default/zero-filled structure that is assigned (copied) to each element of the array.

File: src/game/TacticalAI/FindLocations.cc (hunk 2 above)
Comment (flaviojs) [reply src/game/TacticalAI/FindLocations.cc]: It's a magic debug number.

File: src/game/TacticalAI/AIMain.cc (hunk 3 above)
Comment (flaviojs) [reply src/game/TacticalAI/AIMain.cc]: It's a magic debug number.

File: src/game/Tactical/PathAI.cc (hunk 4 above)
Comment (flaviojs) [reply src/game/Tactical/PathAI.cc]: It's a magic debug number.

**Comment by flaviojs** [comment]
This PR touches too much code so it's easy to make mistakes.
To facilitate review and avoid mistakes, I changed the code to the alternative closest to the original memset.

File: src/game/Laptop/BobbyR.cc (hunk 1 above)
Comment (selaux) [reply src/game/Laptop/BobbyR.cc]: Thanks for clarification, I just wanted to make sure...

**Pull request merged**